### PR TITLE
atomic.conf test test and unit template updates

### PR DESCRIPTION
**Pull request by ashcrow**
## Description
- Updates the unit test template to be in line with other tests and encourage documentation
- Checks the ``atomic.conf`` file to ensure it's valid YAML

**Note** This test is currently expected to fail as the current atomic.conf is NOT valid YAML when the proxy items are checked.

## Related Issue Numbers
- #1004

### CODE REVIEW

File: tests/unit/test_conf.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+    def setUp(self):
+        """
+        Provide the conf file contents freshly for each test.
+        """
+        self.conf = open(os.path.sep.join(['..', '..', 'atomic.conf']), 'r')
+
+    def tearDown(self):
+        """
+        Close the config file after ever test.
```
Comment (TomSweeneyRedHat) [inline tests/unit/test_conf.py]: nit, "ever test" to "every test"

Comment (ashcrow) [inline tests/unit/test_conf.py]: Woops!

File: tests/unit/test_conf.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+
+    def test_config_file_is_valid_yaml_with_items_uncommented(self):
+        """
+        Verifies atomic.conf is valid YAML when examples are uncommented.
+        """
+        # If a command line has a space after it, it's a comment
+        # If a comment line has no space after it, it's an example
+        uncommented = []
+        for line in self.conf.readlines():
+            if line.startswith('#') and len(line) > 2 and line[1] != ' ':
```
Comment (TomSweeneyRedHat) [inline tests/unit/test_conf.py]: Maybe not enough tea yet today, but should it be checking "> 1" rather than "> 2"?   Otherwise wouldn't '#A' get skipped?

Comment (ashcrow) [inline tests/unit/test_conf.py]: The test is just to make sure the line isn't ``#\n`` before checking the second element for a space.

**Comment by ashcrow** [comment]
Looks like ``atomic.conf`` is not where I expect it to be in redhat-ci. Will update.

**Comment by ashcrow** [comment]
Fixed.

**Comment by ashcrow** [comment]
Rebased

**Comment by ashcrow** [comment]
@rhatdan / @baude PTAL

**Comment by rhatdan** [comment]
LGTM

**Comment by giuseppe** [comment]
@rh-atomic-bot r+ 924617e

**Pull request closed without merging**